public/app/features/datasources/components/picker/utils.ts

```diff
@@ -37,3 +38,30 @@ export function dataSourceName(
 
   return 'Unknown';
 }
+
+export function useGetDatasources(filters: GetDataSourceListFilters) {
+  const dataSourceSrv = getDataSourceSrv();
+
+  return (filters: GetDataSourceListFilters) => dataSourceSrv.getList(filters);
+}
+
+export function useGetDatasource() {
+  const dataSourceSrv = getDataSourceSrv();
+
+  const getDataSource = (dataSource: string | DataSourceRef | DataSourceInstanceSettings | null | undefined) => {
+    if (!dataSource) {
+      return undefined;
+    }
+
+    if (typeof dataSource === 'string') {
+      return dataSourceSrv.getInstanceSettings(dataSource);
+    }
+
+    if ('uid' in dataSource) {
+      return dataSourceSrv.getInstanceSettings(dataSource);
+    }
+
+    return undefined;
+  };
+  return getDataSource;
+}
```

Comment:
Something like this
```suggestion
export function useGetDatasources(filters: GetDataSourceListFilters) {
  return dataSourceSrv.getList(filters);
}

export function useGetDatasource(dataSource: string | DataSourceRef | DataSourceInstanceSettings | null | undefined) {
    if (!dataSource) {
      return undefined;
    }

    if (typeof dataSource === 'string') {
      return dataSourceSrv.getInstanceSettings(dataSource);
    }

    if ('uid' in dataSource) {
      return dataSourceSrv.getInstanceSettings(dataSource);
    }

    return undefined;
}
```

Reply:
The reason it's returning a getter is because the result will not update properly when the filtering is done. But if we instead filter the result returned from the hook we can skip the getter :+1: